### PR TITLE
Define folder structure of OSeMOSYS implementations

### PR DESCRIPTION
**Pull request by willu47**
First stab at addressing #56

Adds an `OSeMOSYS_example` folder to the main repository which documents the expected structure of the language implementations. 

@abhishek0208 and @tniet - we can use this pull request as an opportunity to discuss what we think is needed to close issue #56:

- is this an appropriate structure for all language implementations?
- is the information provided in the readme sufficient (at the moment)
- any remaining work should be noted in new issues if it is not added in this pull request

### CODE REVIEW

File: OSeMOSYS_example/README.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+This folder demonstrates the required structure for implementations of the
+OSeMOSYS formulation.
+
+```bash
+OSeMOSYS_example $ tree
+.
+├── README.md
+├── docs
+│   └── index.md
+└── scripts
```
Comment (willu47) [inline OSeMOSYS_example/README.md]: Missing the `src` folder

Comment (abhishek0208) [inline OSeMOSYS_example/README.md:12]: Missing 'src' folder

**Review by abhishek0208 — CHANGES_REQUESTED** [review]
Missing 'src' folder

File: OSeMOSYS_example/README.md (hunk 1 above)
Comment (willu47) [reply OSeMOSYS_example/README.md:12]: Added in commit 495dfc3

**Pull request merged**